tests/rule_based_profiler/parameter_builder/test_parameter_computations.py

```diff
@@ -1,57 +1,341 @@
-from typing import Dict, Optional, Union
+import math
+from numbers import Number
+from typing import Callable, Dict, List, Tuple, Union
 
 import numpy as np
 import pandas as pd
-import scipy.stats as stats
+import scipy
+from packaging import version
 
 from great_expectations.rule_based_profiler.helpers.util import (
-    compute_bootstrap_quantiles,
+    _compute_bootstrap_quantiles_point_estimate_custom_bias_corrected_method,
+    _compute_bootstrap_quantiles_point_estimate_custom_mean_method,
+    _compute_bootstrap_quantiles_point_estimate_scipy_confidence_interval_midpoint_method,
 )
 from great_expectations.rule_based_profiler.parameter_builder.numeric_metric_range_multi_batch_parameter_builder import (
     DEFAULT_BOOTSTRAP_NUM_RESAMPLES,
 )
+from great_expectations.util import probabilistic_test
+from tests.conftest import skip_if_python_below_minimum_version
 
+# Allowable tolerance for how closely a bootstrap method approximates the sample
+EFFICACY_TOLERANCE: float = 1.0e-2
 
-def _generate_distribution_samples(size: Optional[int] = 36) -> pd.DataFrame:
-    data: Dict[str, np.ndarray] = {
-        "normal": np.around(stats.norm.rvs(5000, 1000, size=size)),
-        "uniform": np.around(stats.uniform.rvs(4000, 6000, size=size)),
-        "bimodal": np.around(
-            np.concatenate(
-                [
-                    stats.norm.rvs(4000, 500, size=size // 2),
-                    stats.norm.rvs(6000, 500, size=size // 2),
-                ]
+# Measure of "closeness" between "actual" and "desired" is computed as: atol + rtol * abs(desired)
+# (see "https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html" for details).
+RTOL: float = 1.0e-7
+ATOL: float = 1.0e-2
+
+
+def _compute_quantile_root_mean_squared_error_of_bootstrap(
```

Comment:
@NathanFarmer Any harm making this method public?  Thanks!

Reply:
@alexsherstinsky I see no harm in making these methods public, but for what purpose?

Reply:
@NathanFarmer So that they can be used in other classes/methods...